SU2_CFD/src/iteration/CTurboIteration.cpp

```diff
@@ -64,3 +64,114 @@ void CTurboIteration::InitTurboPerformance(CGeometry* geometry, CConfig** config
   TurbomachineryPerformance = std::make_shared<CTurboOutput>(config, *geometry, *fluid);
   TurbomachineryStagePerformance = std::make_shared<CTurbomachineryStagePerformance>(*fluid);
 }
+
+void CTurboIteration::TurboRamp(CGeometry**** geometry_container, CConfig** config_container, unsigned long iter, unsigned short iZone, unsigned short ramp_flag) {
+  /*--- Generic function for handling turbomachinery ramps ---*/
+  // Grid updates (i.e. rotation/translation) handled seperately to boundary (i.e. pressure/mass flow) updates
+  auto* config = config_container[iZone];
+  auto* geometry = geometry_container[iZone][INST_0][ZONE_0];
+
+  /*--- Lambda function for selecting correct ramp variables ---*/
+  // Ramp coefficient variables
+  auto GetRamp_Coeff = [&](CConfig* &config, unsigned short int x) { 
+    if (ramp_flag == TURBO_RAMP_TYPE::GRID && config->GetRampRotatingFrame()) return config->GetRampRotatingFrame_Coeff(x);
+    else if (ramp_flag == TURBO_RAMP_TYPE::GRID && config->GetRampTranslationFrame()) return config->GetRampTranslationFrame_Coeff(x);
+    else if (ramp_flag == TURBO_RAMP_TYPE::BOUNDARY && config->GetRampOutletPressure()) return config->GetRampOutletPressure_Coeff(x);
+    else if (ramp_flag == TURBO_RAMP_TYPE::BOUNDARY && config->GetRampOutletMassFlow()) return config->GetRampOutletMassFlow_Coeff(x);
+    else return config->GetRampRotatingFrame_Coeff(x); //No ramp specified, does nothing
+  };
```

Comment:
Can you move this logic into CConfig, i.e. pass it the type and flag? So that the interface of CConfig can stay a bit smaller and we don't need to add more function for all the things we may want to ramp.